Add docs for Delete Hooks (#890)
zeit/docs: pages/docs/integrations/v2/integrations-docs-mdx/delete-hooks/integrations-delete-hook-uri.mdx

export const meta = {
  editUrl:
    'pages/docs/integrations/v2/integrations-docs-mdx/delete-hooks/integrations-delete-hook-uri.mdx',
  lastEdited: '2019-06-13T11:13:36.000Z'
}

## Setting Up

> Delete Hook is a HTTP endpoint which is configured to receive HTTP DELETE requests. After you create a such an endpoint in your app, you can add it to the integration via the [Integration Console](https://zeit.co/dashboard/integrations/console).

Reviewer: ```suggestion
The **Delete Hook** is a HTTP endpoint which is configured to receive HTTP [DELETE requests](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/DELETE). After you create the endpoint in your app, you can add it to the Integration via the [Integration Console](https://zeit.co/dashboard/integrations/console).
```

> Inside your integration's page, there's an input to set the "Delete Hook URL". Set the above mentioned HTTP endpoint to that.

Reviewer: ```suggestion
Inside your Integration's configuration page, there is an input to set the **Delete Hook URL**. This is where you should add the HTTP endpoint mentioned above.
```

> Then when the user remove a configuration for your integration, a HTTP DELETE request will send to above mentioned "Delete Hook URL".

Reviewer: ```suggestion
With the Delete Hook URL set, when a user removes a configuration for your Integration, a HTTP DELETE request will be sent to the defined endpoint.
```

> It receive with a JSON encoded HTTP body containing following fields:

Reviewer: ```suggestion
The payload your delete endpoint receives is a JSON encoded HTTP body with the following fields:
```

- `configurationId`
- `userId`
- `teamId` - `null` if the user does not belong to a team

> Take a look at [this example integration](https://github.com/zeit/integrations/tree/master/delete-hooks).

Reviewer: ```suggestion
Take a look at [this example integration](https://github.com/zeit/integrations/tree/master/delete-hooks) to learn more about how Delete Hooks work.
```

## HTTP Response

> You should consider this HTTP request as an event. Once you get the request, you should schedule a task for the cleanup job.

Reviewer: ```suggestion
You should consider this HTTP request as an event. Once you receive the request, you should schedule a task for the cleanup action.
```

> This request has a timeout of 10 seconds. That means, if we did not get a HTTP 200 response within 10 seconds, we will abort the request.

Reviewer: ```suggestion
This request has a timeout of 10 seconds. That means, if a HTTP 200 response is not received within 10 seconds, the request will be aborted.
```

export const meta = {
  editUrl:
    'pages/docs/integrations/v2/integrations-docs-mdx/delete-hooks/integrations-delete-hook-uri.mdx',
  lastEdited: '2019-06-11T23:11:06.000Z'
}

## For Integrations created with Delete Hook URL

> If a Delete Hook URL has been set for an integration, a HTTP `DELETE` request is sent to that URL after the configuration is removed.

Reviewer 2: ```suggestion
When the user removes a configuration, your integration can receive an event. To do that, you need to add a new URL. We call it as the "Delete Hook URL".
You can set it in your integration via the [Integration Console](https://zeit.co/dashboard/integrations/console).

After you set it, we will send a HTTP DELETE request whenever a user removes a configuration. Then you can use that to clean allocated resources for that configuration.
```

Here's a quick example code that demonstrates how this can be tackled:

```js
const micro = require('micro')

module.exports = async function(req, res) {
  if (req.method === 'DELETE') {
    const payload = await micro.json(req)
    console.log(
      'Configuration has been deleted. You can do some cleanup tasks here.'
    )
    console.log(`Payload: ${JSON.stringify(payload)}`)
  }
  return micro.send(res, 200, {})
}
```

List of payload parameters available from the request:

- `configurationId`
- `userId`
- `teamId` - `null` if the user does not belong to a team

> Take a look at the [code on GitHub](https://github.com/zeit/integrations/tree/master/delete-hooks) for a fully working example.

Reviewer 2: ```suggestion
Take a look at the [code on GitHub](https://github.com/zeit/integrations/tree/master/delete-hooks) for a fully working example.

### HTTP Response
You should consider this HTTP request as an event. Once you get the request, you should schedule a task for the cleanup job.

This request has a timeout of 10 seconds. That means, if we did not get a HTTP 200 response within 10 seconds, we will abort the request.
```
